satellite/satellitedb/orders.go

```diff
@@ -266,18 +266,31 @@ func (db *ordersDB) ProcessOrders(ctx context.Context, requests []*orders.Proces
 	// the case where the order has already been processed.  Duplicates and previously
 	// processed orders are rejected
 	for _, request := range requests {
-		insert := "INSERT INTO used_serials (serial_number_id, storage_node_id) SELECT id, ? FROM serial_numbers WHERE serial_number = ?"
+		// create a savepoint so we can rollback if an order has already been used
```

Comment:
nit: I would add to the comment something like

`Avoid the PG error "current transaction is aborted, commands ignored until end of transaction block" if the below insert fails due any constraint. See https://www.postgresql.org/message-id/13131805-BCBB-42DF-953B-27EE36AAF213%40yahoo.com`

I had to figure it out why the solution that I wrote below and after I strikethrough didn't work